### PR TITLE
Implement chained badges

### PR DESCRIPTION
**Pull request by gitsteph**
Hi!  I'm not sure what the status was on this feature; seemed like it was partially implemented then commented out for some reason... submitting this since there weren't updates in the issue on it yet.  :)

A few things I noticed/questions:
* It seems like new `BadgeDefinition`s will only create new `Badge`s for all current `GamificationInterface`s.  If a new `GamificationInterface` is created and saved afterward, that won't receive the `Badge` with prior `BadgeDefinition`s.
* Along the same vein, if we were to later place a `Badge` (linked to a prior `BadgeDefinition`) on a new interface, e.g. by manually creating -> `new_badge = Badge(badge_definition=instance_of_badge_def)`, that `new_badge` won't have fields such as `name`, `description`, etc. that would be populated by `BadgeDefinition` filled.  Is the intent that a `Badge` would only ever be created via `BadgeDefinition`?
* Lastly, your example of awarding a badge points out that you'd grab all the badges then grab the first; that seems like that would always grab the same one though, right?  (Rather than the first one that is not yet acquired)

Finally, thanks for maintaining this!!

(Related to [Issue #3 ](https://github.com/mattjegan/django-gamification/issues/3) )

### CODE REVIEW

File: django_gamification/models.py [hunk 1]
```diff
@@ -60,6 +60,7 @@ def save(self, *args, **kwargs):
                     description=self.description,
                     progression=Progression.objects.create(target=self.progression_target) if self.progression_target
                     else None,
+                    next_badge=self.next_badge,
```
Comment (mattjegan) [inline django_gamification/models.py]: This is the part that looks like it is the trickiest to implement. The value for `next_badge` needs to be an instance of `Badge` but in this context you're assigning a `BadgeDefinition`. Perhaps the best way to accomplish this is to make all the `Badge`s we need to (without `next_badge` set) and then populate that field afterwards in a separate step?

**Comment by mattjegan** [comment]
Thanks for making the PR into the project, I'm always looking for help. Some great points you've mentioned above 😄  You're first point (interfaces not creating missing badges) is definitely a bug which we'll need to address at some point. As for the second point, I did originally intend for `BadgeDefinitions` to be used to create `Badge` instances automatically, do you think this could be improved or perhaps just documented better? Lastly, my example of using `Badge.objects.first()` was intended to demonstrate the use of `Badge.award()` rather than fetching of a particular `Badge`, perhaps this could also be documented better?

File: django_gamification/models.py (hunk 1 above)
Comment (gitsteph) [reply django_gamification/models.py]: Ah, right.  That makes sense; thanks for clarifying.  Just finished editing that part now and extended the `test_models.py`.  I'm trying to figure out how to get the changes I made in my fork's branch to go into this PR now... haha.  (I normally don't use forks, so this is a first for me.)  :)

Comment (gitsteph) [reply django_gamification/models.py]: Ah, nevermind, it's there after all (https://github.com/mattjegan/django-gamification/pull/11/commits) but just not showing in this main view... (at least for me)

**Comment by mattjegan** [comment]
Thanks, looks correct to me 👍  I'll merge it in right now. One thing that I would perhaps recommend in the future is to split your tests into multiple smaller tests. In `test_save`, whenever you think you need another `test_msg#` that would indicate that you may need another `test_` method instead 😄  Anyway, thanks for the awesome contribution and I hope that you may continue to contribute to this project.

**Pull request merged**